os/pm/pm_procfs.c

```diff
@@ -412,29 +413,57 @@ static int power_close(FAR struct file *filep)
 static ssize_t power_read(FAR struct file *filep, FAR char *buffer, size_t buflen)
 {
 	FAR struct power_file_s *priv;
-	ssize_t ret;
-
-	fvdbg("buffer=%p buflen=%d\n", buffer, (int)buflen);
-
-	/* Recover our private data from the struct file instance */
+	size_t totalsize;
+	int domain_id;
+	int last_read;
+	/* Function to copy domain information into buffer */
+	void readprint(const char *format, ...) {
+		size_t copysize;
+		va_list args;
+		va_start(args, format);
+		copysize = vsnprintf(buffer, buflen, format, args);
+		va_end(args);
+		/* Take min of buflen and copysize */
+		if (buflen <= copysize) {
+			copysize = buflen;
+		}
+		/* No need to copy information if we have already read information earlier */
+		if (copysize <= last_read) {
+			last_read -= copysize;
+			return;
+		}
+		/* Overwrite the last read information */
+		if (last_read) {
+			strncpy(buffer, buffer + last_read - 1, copysize - last_read + 1);
+			copysize -= (last_read -1);
+			last_read = 0;
+		}
+		/* Increment the buffer pointer */
+		buflen -= copysize;
+		buffer += copysize;
+		totalsize += copysize;
+		priv->offset += copysize;
+	}
 
 	priv = (FAR struct power_file_s *)filep->f_priv;
-	DEBUGASSERT(priv);
-
-	/* Perform the read based on the directory entry */
-
-	ret = 0;
-
-	if (priv->dir.base.level == 3 && priv->dir.direntry < g_power_direntrycount) {
-		ret = g_power_direntry[priv->dir.direntry].read(filep, buffer, buflen);
```

Comment:
let's keep it

Reply:
Why do we need it ?

Reply:
I thought to apply it considering scalability,
It doesn't seem necessary to keep this.
Your change looks good.